src/partners/changehero.ts

```diff
@@ -101,17 +98,22 @@ export async function queryChangeHero(
 ): Promise<PluginResult> {
   const { settings, apiKeys } = asChangeHeroPluginParams(pluginParams)
   const { apiKey } = apiKeys
-  let { latestTimeStamp, offset, firstAttempt } = settings
+  let offset = 0
+  let { latestIsoDate } = settings
 
   if (typeof apiKey !== 'string') {
-    return { settings: { latestTimeStamp }, transactions: [] }
+    return { settings: { latestIsoDate }, transactions: [] }
   }
 
   const ssFormatTxs: StandardTx[] = []
-  let newLatestTimeStamp = latestTimeStamp
-  // const done = false
+  let previousTimestamp = new Date(latestIsoDate).getTime() - QUERY_LOOKBACK
+  if (previousTimestamp < 0) previousTimestamp = 0
+  const previousLatestIsoDate = new Date(previousTimestamp).toISOString()
+
   try {
-    while (true) {
+    let done = false
+    while (!done) {
+      let oldestIsoDate = '999999999999999999999999999999999999'
```

Comment:
This isn't an ISO date string. Is there some hack going on here that needs explaining with comments?

Reply:
Why was the param changed from `latestTimeStamp` to `latestIsoDate` anyhow; seems like a timestamp is the right data type.

Reply:
I see what's happening here, you're trying to initialize this variable with the largest ISO date. If timestamps were used this would be MAX_SAFE_INTEGER. Timestamps seem more straight forward to use. If the reason for the ISO date is to have a more human-friendly date in the document, then that's fine, but that doesn't mean the implementation must use ISO strings which I feel is hairier to deal with then numbers when doing time-based logic.

Reply:
Fair enough.